src/keymap_manager.hh

```diff
@@ -40,9 +40,8 @@ public:
     KeyList get_mapped_keys(KeymapMode mode) const;
 
     auto get_mapping_keys(Key key, KeymapMode mode) {
-        struct Keys : ConstArrayView<Key> { ScopedSetBool executing; };
         auto& mapping = get_mapping(key, mode);
-        return Keys{mapping.keys, mapping.is_executing};
+        return mapping.keys;
```

Comment:
This is returning the keys by value which means we do a copy here, which removes the need for any trash management. That might be a better approach, I am not sure avoiding the allocation and copy of the mapping will have a significant performance impact.

Reply:
Yeah copying is fine. We could also use a reference-counted pointer